test/e2e/azdiskschedulerextender_test.go

```diff
@@ -43,6 +43,9 @@ var _ = ginkgo.Describe("AzDiskSchedulerExtender", func() {
 	ginkgo.Context("[multi-az]", func() {
 		schedulerExtenderTests(true)
 	})
+	ginkgo.Context("pod-failover-test", func() {
```

Comment:
I wasn't sure where this test should be included, so I kept it in azdiskschedulerextender_test, but can move it to its own file if more appropriate

Reply:
If we want to run it for both v1 and v2 tests, I would again recommend updating the existing failover test in dynamic provisioning rather than adding the new one.